sopel/irc/backends.py

```diff
@@ -80,6 +84,21 @@ def __init__(
         self._keyfile: Optional[str] = keyfile
         self._verify_ssl: bool = verify_ssl
         self._ca_certs: Optional[str] = ca_certs
+        self._ssl_ciphers = ":".join(ssl_ciphers)
+
+        self._ssl_minimum_version = ssl_minimum_version
+        if isinstance(ssl_minimum_version, str):
+            self._ssl_minimum_version = getattr(
```

Comment:
This will confuse mypy/the type hint system, because you redefine what `self._ssl_minimum_version` can be:

* at first it's `Optional[str]`
* then if the getattr succeed, it'll be either "any" or `Optional[TLSVersion]`

Reply:
You also have another option to deal with that:

```python
try:
    self._ssl_minimum_version = ssl.TLSVersion[ssl_minimum_version]
except KeyError:
    LOGGER.error("invalid min version")
    raise ValueError('...')
```

Reply:
Oops. Hm. If I'm reading the docs correctly, could I just use `ValidatedAttribute(parse=...), I think? Then the backend should only ever see the TLSVersion

Reply:
You can absolutely do that, yes.

Reply:
New push seem reasonable? That also fixes the weirdness I mentioned [above](https://github.com/sopel-irc/sopel/pull/2246#issuecomment-1030744713).
If None actually makes it through for ssl_min_ver or ssl_ciphers, `ssl` won't like it, how much do we care about that? Could 'if not None' both of them when they're used...